### ISSUE TITLE
Error 0xd000020c with version 2.1.0.0 and later

### ISSUE THREAD

**Issue by ZupoLlask** [microsoft/WSL#11062]
### Windows Version

Microsoft Windows [Version 10.0.19045.3992]

### WSL Version

2.1.0.0

### Are you using WSL 1 or WSL 2?

- [ ] WSL 2
- [X] WSL 1

### Distro Version

Ubuntu 22.04

### Repro Steps

Download and install MSI files from Github for versions 2.1.0.0 or 2.1.1.0 and try to run `wsl -d Ubuntu`.

### Expected Behavior

Ubuntu should start as usual...

### Actual Behavior

Error: 0xd000020c
Error code: Wsl/Service/CreateInstance/0xd000020c

After reinstall MSI file from Github for version 2.0.15.0, everything works again.
I tried reboot the system with no success.

**Comment by ZupoLlask** [comment]
Here it is...

[WslLogs-2024-01-27_17-15-16.zip](https://github.com/microsoft/WSL/files/14073146/WslLogs-2024-01-27_17-15-16.zip)

**Comment by iourit** [comment]
The error 0xc00000bb can only be returned when the velocity feature 13591025 is not enabled. Please use "stagingtool.exe /enable 13591025". But GPU virtualization in WSL1 is not supported anymore.

**Comment by ZupoLlask** [comment]
@iourit I get this response to "stagingtool.exe /query 13591025"

```
StagingTool [version 10.0.25910.1000 (WinBuild.160101.0800)]

    Feature 13591025 is in its default enablement state for this build
```

Does it help?

**Comment by Sieboldianus** [comment]
> If your WSL version is 2.1.0.0, then you are using WSL 2, not WSL 1.

Can you explain @mbomb007, I could find nowehere that `>2.1.0.0` is limited to WSL2. WSL1 is also not said to be deprecated, so I don't understand.

**Comment by ZupoLlask** [comment]
Yes, my version is >2.1.0.0 and I'm using WSL1. That information is clearly incorrect.

**Comment by qdel** [comment]
Hello,

I am meeting same issue on windows 10 machine and on windows 11 machine.

Is there any workarounds ? Stagingtool call is same.

**Comment by dreamlayers** [comment]
@qdel As a workaround, simply download and install an older .msi and WSL 1 should work afterwards. A reboot is not needed. https://github.com/microsoft/WSL/releases/tag/2.0.14 works for me.

**Comment by qdel** [comment]
> Yes, installing Appx-Package, powershell: `Add-AppxPackage ./Microsoft.WSL_2.0.15.0_x64_ARM64.msixbundle` is a workaround atm.

Should have been more precise, my blame.
Another but downgrading.

It is already what I am doing, but most of my machines will auto update at one point.

Note for everyone reading @Sieboldianus workaround is actually the only working one!

**Comment by benhillis** [comment]
The velocity tag you guys are calling out has never been enabled in external builds. Are you all using internal Windows builds?

**Comment by benhillis** [comment]
@qdel - then I suspect you are running into something different. Can you share logs?

**Comment by qdel** [comment]
Sure: 
[WslLogs-2024-03-06_16-26-04.zip](https://github.com/microsoft/WSL/files/14512013/WslLogs-2024-03-06_16-26-04.zip)

**Comment by benhillis** [comment]
@qdel - thanks, looks the same as the other trace. I'm investigating what's going on here...

**Comment by Rican7** [comment]
I just ran into this myself. Downgrading to [2.0.15.0](https://github.com/microsoft/WSL/releases/tag/2.0.15) also worked for me.

For what it's worth (maybe it'll help?), looking up `subsystem` in my Installed Apps shows a few packages, some pretty old (maybe they're to blame?):

![image](https://github.com/microsoft/WSL/assets/742384/a00961bc-d590-4465-9d4f-6f5a9a2fca99)

**Comment by qdel** [comment]
I found a workaround which works for me it seems:

I exported my wsl1 distribution. This one was originaly based on "Debian" form the microsoft store.
I removed everything mentioning "subsystem" and mostly, Debian., removed also the wsl feature and restarted.

Reinstalled the feature, used wsl --update, imported my wsl1 installation and.... TADA, it worked !
Just add to re-set the default user in /etc/wsl.conf.

I feel like "Debian" from microsoft store is pulling something bad.

**Comment by RAVoigt** [comment]
Downgrading to 2.0.15 fixed this for me as well. No other uninstall/reinstall of distros or anything like that needed.

Microsoft Store background-updated WSL to 2.1.x overnight for me, which broke all of my open instances with the same errors as above.

At a total guess, it looks as though 2.1.x might have issues with (some) distro instances that were created in 2.0.x or earlier?

**Comment by OneBlue** [comment]
Thanks everyone for the info. We're still investigating the root cause this. 

We have some leads but we need a bit more info to validate them, if someone could try the following, that would be super helpful for us: 

- Upgrade to the latest version of WSL via: `wsl.exe --update` 
- Run `wsl --uninstall` To make sure every is cleaned up
- Download [the latest MSI package](https://github.com/microsoft/WSL/releases/download/2.1.4/wsl.2.1.4.0.x64.msi)
- Install it via (elevated command prompt): `msiexec /i <path/to/msi> SKIPLSP=1

Once done, see if the issue is fixed

**Comment by dreamlayers** [comment]
@OneBlue The procedure with updating, uninstalling and installing WSL again did not help me.

It is interesting that I'm able to run Alpine Linux, but not Ubuntu.

**Comment by bjared** [comment]
I ran into the same issue today. I tried the @OneBlue suggestion about upgrading, but I think I was already on that version, and the re-install didn't help.

`WSL version: 2.1.4.0
Kernel version: 5.15.146.1-2
WSLg version: 1.0.60
MSRDC version: 1.2.5105
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.4046`

BUT, what did seem to fix this was installing the Ubuntu (Preview) from the Microsoft Store, because after that installed correctly and Ubuntu 24.04 was functioning, launching the old (22.04) suddenly gave me an "Installing..." message instead of the error of this discussion.  Now both work.  So ... to summarize:

I did:

1. wsl --uninstall
2. installed MSI package via link from @OneBlue 
3. Ran that msiexec command in an elevated command prompt window
4. Rebooted
5. (Still got error)
6. Installed Ubuntu (Preview) from Microsoft Store
7. Verified Ubuntu 24.04 was working
8. Tried Ubuntu 22.04 again, and it no longer gave the error, but began "Installing"

One thing to note, is at #5 in my list, Windows Terminal refused to launch at all, as I had it defaulted to that Ubuntu 22.04 terminal. I changed it to 24.04, because I thought 22.04 was just a lost cause, and it was after that I tried launching an Ubuntu 22.04 tab when I got that "Installing" message in lieu of the error.

I hope this helps narrow down some potential workarounds, or maybe a solution!
